src/vtk_calculator.py

```diff
@@ -85,26 +88,22 @@ def main():
     calc.AddCoordinateScalarVariable("y", 1)
     calc.AddCoordinateScalarVariable("z", 2)
     if args.diff:
-        calc.SetFunction('{}-({})'.format(intag, args.function))
-        logging.info("Evaluated '{}-({})' on the mesh.".format(intag, args.function))
-    else:
+        # Check VTK file has dataname
+        if not vtk_dataset.GetPointData().HasArray(intag):
+            logging.warning("Given mesh has no data for \"{}\".\n ABORTING! \n".format(intag))
+            sys.exit()
+        else:
+            data = v2n(vtk_dataset.GetPointData().GetAbstractArray(intag))
+        # Calculate given function on the mesh
         calc.SetFunction(args.function)
-        logging.info("Evaluated '{}' on the mesh.".format(args.function))
-    calc.SetResultArrayName(args.tag)
-    calc.Update()
-    logging.info("Evaluated function saved to {} variable".format(args.tag))
-
-    if args.diff:
-
-        # Transfer Data to Numpy Array to Compute Statistics
-        difference = []
-        diffArr = calc.GetOutput().GetPointData().GetAbstractArray(args.tag)
-        num_points = vtk_dataset.GetNumberOfPoints()
-        for i in range(num_points):
-            difference.append(diffArr.GetTuple1(i))
-        difference = np.array(difference)
+        calc.SetResultArrayName("function")
+        calc.Update()
+        func = v2n(calc.GetOutput().GetPointData().GetAbstractArray("function"))
+        difference = data - func
+        logging.info("Evaluated \"{}\"-\"({})\" on the mesh.".format(intag, args.function))
```

Comment:
```suggestion
        logging.info("Evaluated \"{}\"-\"({})\" (data - function) on the mesh.".format(intag, args.function))
```
Maybe also add the related mesh name here.

Reply:
> This is so nice, thank you very much @kursatyurt. Only some minor logging things to consider. My cases seem to work now as expected (at least software-wise). I will probably try some convergence studies within the next days so that I will test the whole pipeline.
> 
> Feel free to merge, when you are ready.

Thank you for the quick review.